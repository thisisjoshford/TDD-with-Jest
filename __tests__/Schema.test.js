  
const { Schema } = require('../lib/Schema.js');

describe('Schema testing', () => {

  const schema = new Schema({
    name: {
      type: String,
      required: true
    },
    age: {
      type: Number,
      required: true
    },
    weight: {
      type: String
    }
  });

  const spot = {
    name: 'spot',
    age: 5,
    weight: '20 lbs'
  };

  const rover = {
    name: 'rover',
    age: '10'
  };

  const molly = {
    age: 'wouldnt you like to know'
  };

  const grover = {
    name: 'grover',
    weight: '50 lbs'
  };

  const badDog = {
    name: {},
    age: 1,
    weight: '50 lbs'
  };

  // it('Returns spot as is', () => {
  //   expect(schema.validate(spot)).toEqual(spot);
  // });

  // it('Returns rover with the age turned into a number', () => {
  //   expect(schema.validate(rover)).toEqual({
  //     name: 'rover',
  //     age: 10
  //   });
  // });

  it('Throws an error for an dog missing name', () => {
    expect(() => schema.validate(molly)).toThrowErrorMatchingSnapshot();
  });

  it('Throws an error for a dog missing an age', () => {
    expect(() => schema.validate(grover)).toThrowErrorMatchingSnapshot();
  });

  it('Throws an error for a dog with a name that has an invalid data type', () => {
    expect(() => schema.validate(badDog)).toThrowErrorMatchingSnapshot();
  });
});
