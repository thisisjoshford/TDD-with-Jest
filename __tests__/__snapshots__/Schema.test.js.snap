// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema testing Throws an error for a dog missing an age 1`] = `"undefined is required"`;

exports[`Schema testing Throws an error for a dog with a name that has an invalid data type 1`] = `"Cannot cast >>[object Object]<< to String"`;

exports[`Schema testing Throws an error for an dog missing name 1`] = `"undefined is required"`;
